presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -1388,6 +1517,14 @@ private static void throwDivisionByZeroException()
         throw new ArithmeticException("Division by zero");
     }
 
+    private static void multiplyShiftDestructive(Slice decimal, Slice multiplier, int rightShifts)
```

Comment:
That's a micro optimization for some edge cases like:
```
static void shiftLeftDestructive(Slice decimal, int leftShifts)
{
    if (leftShifts == 0) {
        return;
    }
    (...)
}
```
This avoids unnecessary data coping and packing values in the slice.